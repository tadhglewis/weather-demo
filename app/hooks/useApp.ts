import { useContext } from "react";
import AppContext from "../context/AppContext";

export default () => useContext(AppContext);
