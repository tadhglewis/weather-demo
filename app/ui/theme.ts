import theme from "styled-theming";

export const primary = theme("theme", {
  light: "#49b0cf",
  dark: "#354d71",
});

export const secondary = theme("theme", {
  light: "#1e1e1e",
  dark: "white",
});

export const background = theme("theme", {
  light: "white",
  dark: "#1e1e1e",
});

export const textPrimary = theme("theme", {
  light: "white",
  dark: "white",
});

export const textSecondary = theme("theme", {
  light: "#fafafa",
  dark: "#fafafa",
});
