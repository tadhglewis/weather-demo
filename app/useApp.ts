import { useEffect, useState } from "react";
import * as Location from "expo-location";
import { Alert } from "react-native";

export default () => {
  const [location, setLocation] = useState<
    | {
        longitude: number;
        latitude: number;
      }
    | undefined
  >(undefined);

  useEffect(() => {
    (async () => {
      let { status } = await Location.requestPermissionsAsync();

      if (status !== "granted") {
        Alert.alert("Permission to access location was denied");
      }

      let { coords } = await Location.getCurrentPositionAsync({});
      setLocation({ longitude: coords.longitude, latitude: coords.latitude });
    })();
  }, []);

  return { location };
};
