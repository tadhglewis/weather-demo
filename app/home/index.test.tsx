import React from "react";
import { fireEvent, render } from "@testing-library/react-native";
import "jest-styled-components";
import theme from "styled-theming";
import Home from "./";
import { StackScreenProps } from "@react-navigation/stack";
import useApp from "../hooks/useApp";

jest.mock("../hooks/useApp");

jest.mock("@fortawesome/react-native-fontawesome", () => ({
  FontAwesomeIcon: "",
}));

// mocking and always return light theme due to a bug where ThemeProvider isn't passing in the theme to styled-theming in tests
jest.mock("styled-theming", () => (name: string, values: theme.ThemeMap) =>
  values.light
);

describe("home", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should switch to dark theme", () => {
    const mockSetTheme = jest.fn();

    (useApp as jest.Mock).mockReturnValue({
      setTheme: mockSetTheme,
      theme: "light",
    });

    const { getByTestId } = render(
      <Home
        {...({ route: { params: { location: undefined } } } as StackScreenProps<
          AppNav,
          "home"
        >)}
      />
    );

    fireEvent.press(getByTestId("lightsOut"));

    expect(mockSetTheme).toHaveBeenCalledWith("dark");
  });
});
