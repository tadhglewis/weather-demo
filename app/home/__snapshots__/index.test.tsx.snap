// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home should switch to dark theme 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      Object {
        "height": 1334,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginTop": 16,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#49b0cf",
            "borderRadius": 16,
            "elevation": 3,
            "marginBottom": 0,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 0,
            "opacity": 1,
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 8,
            "shadowColor": "#49b0cf",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.22,
            "shadowRadius": 2.22,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
              },
            ]
          }
        >
          Melbourne
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#49b0cf",
            "borderRadius": 16,
            "elevation": 3,
            "marginBottom": 0,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 0,
            "opacity": 1,
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 8,
            "shadowColor": "#49b0cf",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.22,
            "shadowRadius": 2.22,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
              },
            ]
          }
        >
          Brisbane
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#49b0cf",
            "borderRadius": 16,
            "elevation": 3,
            "marginBottom": 0,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 0,
            "opacity": 1,
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 8,
            "shadowColor": "#49b0cf",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.22,
            "shadowRadius": 2.22,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
              },
            ]
          }
        >
          London
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": 0,
        "opacity": 1,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
        "position": "absolute",
      }
    }
    testID="lightsOut"
  >
    <
      icon={
        Array [
          "far",
          "lightbulb",
        ]
      }
      size={32}
      style={
        Array [
          Object {
            "color": "#1e1e1e",
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`home should switch to dark theme 2`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
      Object {
        "height": 1334,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "marginTop": 16,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#49b0cf",
            "borderRadius": 16,
            "elevation": 3,
            "marginBottom": 0,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 0,
            "opacity": 1,
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 8,
            "shadowColor": "#49b0cf",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.22,
            "shadowRadius": 2.22,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
              },
            ]
          }
        >
          Melbourne
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#49b0cf",
            "borderRadius": 16,
            "elevation": 3,
            "marginBottom": 0,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 0,
            "opacity": 1,
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 8,
            "shadowColor": "#49b0cf",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.22,
            "shadowRadius": 2.22,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
              },
            ]
          }
        >
          Brisbane
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#49b0cf",
            "borderRadius": 16,
            "elevation": 3,
            "marginBottom": 0,
            "marginLeft": 8,
            "marginRight": 8,
            "marginTop": 0,
            "opacity": 1,
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 8,
            "shadowColor": "#49b0cf",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.22,
            "shadowRadius": 2.22,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
              },
            ]
          }
        >
          London
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "bottom": 0,
        "opacity": 1,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
        "position": "absolute",
      }
    }
    testID="lightsOut"
  >
    <
      icon={
        Array [
          "far",
          "lightbulb",
        ]
      }
      size={32}
      style={
        Array [
          Object {
            "color": "#1e1e1e",
          },
        ]
      }
    />
  </View>
</View>
`;
